substrate-archive-backend/src/runtime_version_cache.rs

```diff
@@ -177,8 +177,8 @@ impl<B: BlockT> VersionRange<B> {
 		Self { start: *first.block.header().number(), end: *last.block.header().number(), version }
 	}
 
-	fn contains_block(&self, b: NumberFor<B>) -> bool {
-		(b > self.start && b < self.end) || b == self.start || b == self.end
+	fn contains_block(&self, b: &NumberFor<B>) -> bool {
+		(self.start..=self.end).contains(b)
```

Comment:
Does using the range have space or time complexity overhead relative to boolean logic?

What not use `b >= self.start && b <= self.end`?

Reply:
Under the hood they should be exactly the same. 
The idea here is if someone someday figures out a way to optimize `b >= self.start && b <= self.end` we'll get that optimization for free as soon as it lands in the stdlib/compiler. Likely moot in this case, but you never know!

Reply:
(It also makes clippy STFU)